src/mds/CDir.cc

```diff
@@ -2204,6 +2213,9 @@ void CDir::_omap_fetched(bufferlist& hdrbl, map<string, bufferlist>& omap,
 	dout(12) << "_fetched got null for key " << key << ", added " << *dn << dendl;
       }
       mdcache->touch_dentry(dn);
+
+      if (!(++count % 2000))
```

Comment:
mdcache->mds->heartbeat_reset_grace(2) ?

Reply:
Yeah, will fix it. Thanks.